test/Microsoft.ML.Tests/ScenariosWithDirectInstantiation/TensorflowTests.cs

```diff
@@ -74,6 +74,103 @@ public void TensorFlowTransformMatrixMultiplicationTest()
             }
         }
 
+        private class ShapeData
+        {
+            // Data will be passed as flat vector.
+            // Intended data shape [2,2], model shape [2, None]
+            [VectorType(4)]
+            public float[] TwoDim;
+
+            // Data will be passed as 3-D vector.
+            // Intended data shape [1, 2, 2], model shape [1, None, 2]
+            [VectorType(1, 2, 2)]
+            public float[] ThreeDim;
+
+            // Data will be passed as flat vector.
+            // Intended data shape [1, 2, 2, 3], model shape [1, None, None, 3]
+            [VectorType(12)]
+            public float[] FourDim;
+
+            // Data will be passed as 3-D vector.
+            // Intended data shape [2, 2, 2, 2], model shape [2, 2, 2, 2]
+            [VectorType(2, 2, 2, 2)]
+            public float[] FourDimKnown;
+        }
+
+        private List<ShapeData> GetShapeData()
+        {
+            return new List<ShapeData>(new ShapeData[] {
+                        new ShapeData() {   TwoDim = new[] { 1.0f, 2.0f, 3.0f, 4.0f },
+                                            ThreeDim = new[] { 11.0f, 12.0f, 13.0f, 14.0f },
+                                            FourDim = new[]{ 21.0f, 22.0f, 23.0f, 24.0f, 25.0f, 26.0f,
+                                                             27.0f, 28.0f, 29.0f, 30.0f, 31.0f, 32.0f },
+                                            FourDimKnown = new[]{ 41.0f , 42.0f, 43.0f, 44.0f, 45.0f, 46.0f, 47.0f, 48.0f,
+                                                                  49.0f , 50.0f, 51.0f, 52.0f, 53.0f, 54.0f, 55.0f, 56.0f}
+                                        },
+                        new ShapeData() { TwoDim = new[] { 101.0f, 102.0f, 103.0f, 104.0f },
+                                            ThreeDim = new[] { 111.0f, 112.0f, 113.0f, 114.0f },
+                                            FourDim = new[]{ 121.0f, 122.0f, 123.0f, 124.0f, 125.0f, 126.0f,
+                                                             127.0f, 128.0f, 129.0f, 130.0f, 131.0f, 132.0f},
+                                            FourDimKnown = new[]{ 141.0f , 142.0f, 143.0f, 144.0f, 145.0f, 146.0f, 147.0f, 148.0f,
+                                                                  149.0f , 150.0f, 151.0f, 152.0f, 153.0f, 154.0f, 155.0f, 156.0f }
+                                        }
+                });
+        }
+
+        [ConditionalFact(typeof(Environment), nameof(Environment.Is64BitProcess))] // TensorFlow is 64-bit only
+        public void TensorFlowTransformInputShapeTest()
+        {
+            var modelLocation = "model_shape_test";
```

Comment:
I assume `model_shape_test` has both unknown input and output dimensions, correct?

Reply:
Yes, I described it on line 77-98

---
In reply to: [255145486](https://github.com/dotnet/machinelearning/pull/2475#discussion_r255145486) [](ancestors = 255145486)